src/Language/GraphQL/Draft/Parser.hs

```diff
@@ -479,3 +488,41 @@ between open close p = tok open *> p <* tok close
 -- `empty` /= `pure mempty` for `Parser`.
 optempty :: Monoid a => Parser a -> Parser a
 optempty = option mempty
+
+-- | Parses strings delimited by triple quotes. 
+-- http://spec.graphql.org/June2018/#sec-String-Value
+blockString :: Parser Text
+blockString = do
+  _ <- tripleQuotes <?> "opening triple quotes"
+  lines_ <- T.lines . T.pack <$> AT.manyTill AT.anyChar tripleQuotes <?> "the body of a triple quoted string"
+  let headline = if lines_ == [] then "" else head lines_
+  let tail_ = drop 1 lines_ -- not tail
+  let commonIndentation = foldr min maxBound (countIndentation <$> tail_)
+  let rlines = foldr (removeCommonIndentation commonIndentation) Nothing tail_
+  return $ case rlines of
+    Nothing -> headline
+    Just reformatedLines -> rebuild (sanitize $ headline:reformatedLines)
+ where
+  rebuild :: [Text] -> Text
+  rebuild = fromMaybe "" . fmap fst . T.unsnoc . T.unlines
+  sanitize :: [Text] -> [Text]
+  sanitize = dropWhileEnd' onlyWhiteSpace  . dropWhile onlyWhiteSpace
+  onlyWhiteSpace :: Text -> Bool
+  onlyWhiteSpace t = T.null t || T.all ws t
+  removeCommonIndentation :: Int -> Text -> Maybe [Text] -> Maybe [Text] 
+  removeCommonIndentation smallest a x = 
+    let new = T.drop smallest a
+     in case x of
+      Nothing -> Just [new]
+      Just acc -> Just $ new:acc
+  countIndentation :: Text -> Int
+  countIndentation = fromMaybe maxBound . T.findIndex (not . ws)
+  tripleQuotes = AT.string "\"\"\""
+
+-- whitespace
+ws :: Char -> Bool
+ws c = c == ' ' || c == '\t'
+
+-- copied from https://hackage.haskell.org/package/extra-1.7.9/docs/src/Data.List.Extra.html
+dropWhileEnd' :: (a -> Bool) -> [a] -> [a]
+dropWhileEnd' p = foldr (\x xs -> if null xs && p x then [] else x : xs) []
```

Comment:
There's a minor question of copyright we have to figure out before this PR is merged. Please don't resolve this comment until we figure out if this is OK.